### ISSUE TITLE
Nuxt SSR Issue

### ISSUE THREAD

**Issue by AmareshKulkarni** [antoine92190/vue-advanced-chat#204]
I have tried to folllow the solution suggested in (#175) but still, I am unable to make it work. I am getting the following error.

```
client.js:103 TypeError: Cannot read property 'addEventListener' of undefined
    at VueComponent.mounted (vue-advanced-chat.common.js:37045)
    at invokeWithErrorHandling (vue.runtime.esm.js:1854)
    at callHook (vue.runtime.esm.js:4219)
    at Object.insert (vue.runtime.esm.js:3139)
    at invokeInsertHook (vue.runtime.esm.js:6346)
    at VueComponent.patch2 [as __patch__] (vue.runtime.esm.js:6565)
    at VueComponent.Vue2._update (vue.runtime.esm.js:3948)
    at VueComponent.updateComponent (vue.runtime.esm.js:4060)
    at Watcher2.get (vue.runtime.esm.js:4479)
    at Watcher2.run (vue.runtime.esm.js:4554)
```

Offending line of code seems to be the addEventListener below in **vue-advanced-chat.common.js**:
```

mounted: function mounted() {
    var _this3 = this;

    this.newMessages = [];
    var isMobile = detectMobile();
    this.$refs.roomTextarea.addEventListener('keyup', debounce(function (e) {
      if (e.key === 'Enter' && !e.shiftKey && !_this3.fileDialog) {
        if (isMobile) {
          _this3.message = _this3.message + '\n';
          setTimeout(function () {
            return _this3.onChangeInput();
          });
        } else {
          _this3.sendMessage();
```

My Component code is :

```
<template>
  <client-only>
    <chat-window
      :current-user-id="currentUserId"
      :show-add-room="false"
      :show-audio="false"
      :messages-loaded="true"
      :rooms-loaded="true"
      :rooms="rooms"
      :messages="messages"
    />
  </client-only>
</template>

<script>
import "vue-advanced-chat/dist/vue-advanced-chat.css";
const importChat = () => {
  if (process.client) {
    return { ChatWindow: () => import("vue-advanced-chat") };
  }
  return {};
};
export default {
  components: importChat(),
  data() {
    return {
      rooms: [],
      messages: [],
      currentUserId: 1234,
    };
  },
};
</script>
```

I have not used plugin config or anything else

**Comment by antoine92190** [comment]
That's strange, I created a nuxt project and I don't have any issue.
Could you clone the repo and try out the `nuxt` branch, see if it works for you?

**Comment by AmareshKulkarni** [comment]
Thank you @antoine92190. I cloned and see the application comes up fine. However, when I copy the content of the index.vue into my application, I see weird behavior. Need to research further what if there are any conflicts. I will post what I find. But if something rings a bell based on the below screenshot, please let me know.

![image](https://user-images.githubusercontent.com/1945114/132966462-543a293e-30c0-420d-a1b8-de2a2c373c6c.png)

**Comment by AmareshKulkarni** [comment]
Finally was able to reproduce this with the nuxt branch. Can't see the logic but here is what I ended up doing to reproduce this:

1. I was using Babel and there was a .babelrc in the root of my project. Copied that into the nuxt branch cloned from above.
2. I was using Yarn for dependency management and used that for installing dependencies and serving the app
3. With the above 2 settings, in the nuxt branch, I was getting exactly the same empty screen as above.
4. Now I reverted my changes with git reset --hard and started to serve that again with yarn, but the issue did not resolve.
5. For some reason, once the build was messed up with .babel config that I had, even after resetting to the branch, I could not get past that empty screen
6. Switched to npm and it started to work without any issue

I might be doing something very silly, but wanted to share what I ran into .. Thank you for your help @antoine92190
